grpclb/src/main/java/io/grpc/grpclb/GrpclbState.java

```diff
@@ -740,34 +823,45 @@ public boolean equals(Object other) {
     }
   }
 
-  private interface RoundRobinEntry {
+  @VisibleForTesting
+  interface RoundRobinEntry {
     PickResult picked(Metadata headers);
   }
 
   @VisibleForTesting
   static final class BackendEntry implements RoundRobinEntry {
+    final Subchannel subchannel;
```

Comment:
Why is this being stored now?

Reply:
Because it's always non-null in `BackendEntry`, and there are quite a few places that need to retrieve the subchannel from `BackendEntry`. `entry.subchannel` is easier to reason than `entry.result.getSubchannel()` that you won't get a `null`.